### DISCUSSION TITLE
Possible to turn off iCloud sync?

### DISCUSSION THREAD

**Discussion by jayhawx** [MKT-Playbook/Hub#50] [Q&A] [answered]
I noticed that the app takes up quite a bit of space in my iCloud storage. But I don't notice an option to turn off iCloud sync in the app. I only want it stored locally on my phone. Is it possible to do that?

**Comment by TitusPosiadly** [comment]
Hey there! Yes, you can turn off (and delete) iCloud storage through the system settings.

Just follow this guide by Apple:
https://support.apple.com/en-mide/118225

To delete the MKT Playbook app data from your iCloud, you have to tap on "Storage" instead of "Saved to iCloud", locate the app, and tap on "Delete data from iCloud"

Before you do any of this, I suggest you export a CSV file to back up your inventory through the app. Just select the MKT Toolbox preset and save the file to your device storage.

Out of curiosity, how much data did the app use in your iCloud?

**Reply by jayhawx** [reply]
Ok thanks! 231mb.

**Comment by jayhawx** [comment]
> Hey there! Yes, you can turn off (and delete) iCloud storage through the system settings.
> 
> 
> 
> Just follow this guide by Apple:
> 
> https://support.apple.com/en-mide/118225
> 
> 
> 
> To delete the MKT Playbook app data from your iCloud, you have to tap on "Storage" instead of "Saved to iCloud", locate the app, and tap on "Delete data from iCloud"
> 
> 
> 
> Before you do any of this, I suggest you export a CSV file to back up your inventory through the app. Just select the MKT Toolbox preset and save the file to your device storage.
> 
> 
> 
> Out of curiosity, how much data did the app use in your iCloud?

Just making sure if I delete the iCloud storage option that it doesn't deleted in the app?

I went into the system settings as well just to ensure there wasn't an iCloud feature in there. Nothing, but that might be expected

![image](https://github.com/user-attachments/assets/99f5cbf4-1b87-4eca-baa7-268d4fbcb80d)

**Comment by TitusPosiadly** [comment]
As long as you disable iCloud syncing first, it should be fine. You might don't even have to delete the iCloud data if Apple automatically removes it when you disconnect the app, but I'm not so sure about that. If you want to be safe just disable iCloud without deleting the data manually. And like I said, I would suggest you make a backup first via the CSV exporter just to be safe.

The settings for iCloud aren't in the app setting but rather in the system-wide iCloud settings:

> 1. Go to Settings > your name.
> 2. Tap iCloud.
> 3. Tap See All next to Saved to iCloud. In iOS 17 or earlier, tap Show All under Apps Using iCloud. Tap an app to choose whether to sync or use it with your device or not. 

Hope this helps and let me know if anything else is unclear.

**Comment by jayhawx** [comment]
> As long as you disable iCloud syncing first, it should be fine. You might don't even have to delete the iCloud data if Apple automatically removes it when you disconnect the app, but I'm not so sure about that. If you want to be safe just disable iCloud without deleting the data manually. And like I said, I would suggest you make a backup first via the CSV exporter just to be safe.
> 
> 
> 
> The settings for iCloud aren't in the app setting but rather in the system-wide iCloud settings:
> 
> 
> 
> > 1. Go to Settings > your name.
> 
> > 2. Tap iCloud.
> 
> > 3. Tap See All next to Saved to iCloud. In iOS 17 or earlier, tap Show All under Apps Using iCloud. Tap an app to choose whether to sync or use it with your device or not. 
> 
> 
> 
> Hope this helps and let me know if anything else is unclear.

When I disable iCloud that way and go back into the app my cards are gone though. 

![image](https://github.com/user-attachments/assets/4824af52-a12e-45e3-bbba-88f74f2a2ab3)

**Comment by TitusPosiadly** [comment, marked as answer]
Ok, that should definitely not happen, but you are right, I just tested it myself and had the same issue. As long as you didn't delete the iCloud data yet, you should at least be able to get your data back for the time being (although you might have to wait a couple of minutes with the app open and run the duplicate cards checker after you get some cup recommendations again through the app settings to get everything back to normal).

Normally the app should save all your data to the device AND the iCloud but it looks like the provided framework from Apple only saves it to the cloud for some reason. I have to look into it, but it looks like it's currently in fact, and unfortunately not possible to disable iCloud, sorry. 

And very sorry for the inconvenience. Like I said I will look into it and will try to find a way to properly disable iCloud.

I will convert this discussion to a new issue and will keep you posted on my progress. And let me know if you need any more assistance with getting your data back.

**Reply by jayhawx** [reply]
Ok thanks!

**Comment by TitusPosiadly** [comment]
Btw I think I actually already figured out the issue and a solution at least for the cards disappearing when iCloud gets disabled. But apparently when you try to delete the iCloud data through the system settings the iPhone can sometimes delete the iCloud AND the device storage, so I'm trying to find a way to significantly reduce the iCloud data size manually through the app. This way you can reduce the data size first and disable iCloud after, to keep the used iCloud storage as small as possible. I will try to get everything ready in the next couple of days and will hopefully get an app update ready sometime next week.

**Comment by TitusPosiadly** [comment]
Hey, first of all, sorry for the delay. Some work came up that took priority, but I finally pushed an update to the AppStore today that fixes the issue with disappearing data when turning off iCloud. I also added an option in the app settings to temporarily delete all thumbnails, which should reduce the data stored in iCloud quite a bit. So if you want to free up some iCloud storage you should first delete the thumbnails and then disable iCloud through the system settings (just like in the guide I posted above). The thumbnails will be regenerated the next time you open up your card list. Let me know if you run into any issues.
